src/containers/shared/SocketContext.ts

```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import { XrplClient } from 'xrpl-client'
+
+const LOCALHOST_URLS = ['localhost', '127.0.0.1', '0.0.0.0']
+
+function isInsecureWs(rippledHost: string | null): boolean {
+  return (
+    !!Number(process.env.REACT_APP_INSECURE_WS) ||
+    LOCALHOST_URLS.some((url) => rippledHost?.includes(url)) ||
+    rippledHost === ''
+  )
+}
+
+function getSocket(rippledUrl: string): XrplClient {
+  const rippledHost = rippledUrl ?? process.env.REACT_APP_RIPPLED_HOST
+  const prefix = isInsecureWs(rippledHost) ? 'ws' : 'wss'
+  const wsUrls: string[] = []
+  if (rippledHost.includes(':')) {
+    wsUrls.push(`${prefix}://${rippledHost}`)
+  } else {
+    wsUrls.push.apply(wsUrls, [
+      `${prefix}://${rippledHost}:${process.env.REACT_APP_RIPPLED_WS_PORT}`,
+      `${prefix}://${rippledHost}:443`,
+    ])
+  }
+  const socket = new XrplClient(wsUrls)
+  const hasP2PSocket =
+    process.env.REACT_APP_P2P_RIPPLED_HOST != null &&
+    process.env.REACT_APP_P2P_RIPPLED_HOST !== ''
+  // @ts-ignore - will be removed eventually
```

Comment:
When will this be removed and why? (Just so future readers can tell if this still makes sense to ignore)

Reply:
It will be removed when we support clio only.